build.sh

```diff
@@ -0,0 +1,176 @@
+HELP="$0 [<target> ...] [<flag> ...]
+ where <target> is:
+   clean             - remove all existing build artifacts and configuration (start over)
+   libcuml           - build the cuml C++ code only
+   cuml              - build the cuml Python package
+   prims             - build the ML prims tests
+ and <flag> is:
+   -v                - verbose build mode
+   -g                - build for debug
+   -n                - no install step
+   --buildAllGPUArch - build for all supported GPU architectures
+   --multigpu        - Build cuml with multigpu support (requires libcumlMG and CUDA >=10.0)
+   -h                - print this text
+
+ default action (no args) is to build and install 'libcuml', 'cuml', and 'prims' targets only for the detected GPU arch
+"
+LIBCUML_BUILD_DIR=${REPODIR}/cpp/build
+CUML_BUILD_DIR=${REPODIR}/python/build
+BUILD_DIRS="${LIBCUML_BUILD_DIR} ${CUML_BUILD_DIR}"
+
+# Set defaults for vars modified by flags to this script
+VERBOSE=""
+BUILD_TYPE=Release
+INSTALL_TARGET=install
+BUILD_ALL_GPU_ARCH=0
+MULTIGPU=""
+
+# Set defaults for vars that may not have been defined externally
+#  FIXME: if INSTALL_PREFIX is not set, check PREFIX, then check
+#         CONDA_PREFIX, but there is no fallback from there!
+INSTALL_PREFIX=${INSTALL_PREFIX:=${PREFIX:=${CONDA_PREFIX}}}
+PARALLEL_LEVEL=${PARALLEL_LEVEL:=""}
+BUILD_ABI=${BUILD_ABI:=ON}
+
+function hasArg {
+    (( ${NUMARGS} != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
+}
+
+if hasArg -h || hasArg --help; then
+    echo "${HELP}"
+    exit 0
+fi
+
+# Check for valid usage
+if (( ${NUMARGS} != 0 )); then
+    for a in ${ARGS}; do
+	if ! (echo " ${VALIDARGS} " | grep -q " ${a} "); then
+	    echo "Invalid option: ${a}"
+	    exit 1
+	fi
+    done
+fi
+
+# Process flags
+if hasArg -v; then
+    VERBOSE=1
+fi
+if hasArg -g; then
+    BUILD_TYPE=Debug
+fi
+if hasArg -n; then
+    INSTALL_TARGET=""
+fi
+if hasArg --buildAllGPUArch; then
+    BUILD_ALL_GPU_ARCH=1
+fi
+if hasArg --multigpu; then
+    MULTIGPU=--multigpu
+fi
+
+# Various build options use nvidia-smi for querying the system
+HAS_NVIDIA_SMI=1
+which nvidia-smi > /dev/null
+if (( $? != 0 )); then
+    HAS_NVIDIA_SMI=0
+fi
+
+# If clean given, run it prior to any other steps
+if hasArg clean; then
+    # If the dirs to clean are mounted dirs in a container, the
+    # contents should be removed but the mounted dirs will remain.
+    # The find removes all contents but leaves the dirs, the rmdir
+    # attempts to remove the dirs but can fail safely.
+    for bd in ${BUILD_DIRS}; do
+	if [ -d ${bd} ]; then
+	    find ${bd} -mindepth 1 -delete
+	    rmdir ${bd} || true
+	fi
+    done
+fi
+
+################################################################################
+# Configure for building all C++ targets
+if (( ${NUMARGS} == 0 )) || hasArg libcuml || hasArg prims; then
+
+    # Configure the build for the GPU type of this machine, or all (GPU_ARCH="")
```

Comment:
Perhaps we can add an issue to add more GPUs that can be detected in the future? This only detects the ones that we use in CI, but nothing else

Reply:
I filed issue #628 and added a TODO